src/main/io/gps.c

```diff
@@ -686,77 +1100,144 @@ void gpsInitUblox(void)
+                        }
+                        break;
+                    case UBLOX_MSG_POSLLH:
+                        if (gpsData.ubloxM7orAbove) {
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_POSLLH, 0);
+                        } else {
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_POSLLH, 1);
                         }
                         break;
                     case UBLOX_MSG_STATUS:
-                        if (gpsData.ubloxUsePVT) {
-                            gpsData.state_position++;
+                        if (gpsData.ubloxM7orAbove) {
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_STATUS, 0);
                         } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_STATUS, 1); // set STATUS MSG rate
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_STATUS, 1);
                         }
                         break;
                     case UBLOX_MSG_VELNED:
-                        if (gpsData.ubloxUsePVT) {
-                            gpsData.state_position++;
+                        if (gpsData.ubloxM7orAbove) {
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_VELNED, 0);
                         } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_VELNED, 1); // set VELNED MSG rate
+                            ubloxSetMessageRate(CLASS_NAV, MSG_NAV_VELNED, 1);
                         }
                         break;
-                    case UBLOX_MSG_SATINFO:
-                        if (gpsData.ubloxUseSAT) {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_SAT, 5); // set SAT MSG rate (every 5 cycles)
+                    case UBLOX_MSG_DOP:
+                        if (gpsData.ubloxM7orAbove) {
+                            if (gpsData.ubloxM9orAbove) {
+                                ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_DOP_UART1, 0);
+                            } else {
+                                ubloxSetMessageRate(CLASS_NAV, MSG_NAV_DOP, 0);
+                            }
                         } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_SVINFO, 5); // set SVINFO MSG rate (every 5 cycles)
+                            if (gpsData.ubloxM9orAbove) {
+                                ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_DOP_UART1, 1);
+                            } else {
+                                ubloxSetMessageRate(CLASS_NAV, MSG_NAV_DOP, 1);
+                            }
                         }
                         break;
-                    case UBLOX_MSG_DOP:
-                        ubloxSetMessageRate(CLASS_NAV, MSG_DOP, 1); // set DOP MSG rate
+                    case UBLOX_SAT_INFO:
+                        // disable all sat info on power up, enabling while receiving data if configurator is connected
+                        if (gpsData.ubloxM8orAbove) {
```

Comment:
DRY: you already have a `requestSatelliteInfo()`, use that instead

Reply:
The code has been refactored, please review the changes.